src/store/modules/purchaseOrder/actions.ts

```diff
@@ -11,19 +11,23 @@ import emitter from '@/event-bus'
 const actions: ActionTree<PurchaseOrderState, RootState> = {
 
   // Find Product
-  async findPurchaseOrders ({ commit, state }) {
+  async findPurchaseOrders ({ commit, state }, payload) {
 
     // Show loader only when new query and not the infinite scroll
     // if (payload.viewIndex === 0) emitter.emit("presentLoader");
 
     let resp;
 
     try {
-      resp = await PurchaseOrdersService.fetchPurchaseOrders()
+      resp = await PurchaseOrdersService.fetchPurchaseOrders(payload)
 
       // resp.data.response.numFound tells the number of items in the response
+      const orders = resp.data.grouped.orderId.groups
```

Comment:
If the resp does not contain grouped then this line will through an error of `cannot read property grouped of undefined`. So, we will assign the resp to a variable after we have checked that the status is 200 and we are getting some data.

Reply:
Improved code quality Sir.